### PR TITLE
Recognise both byte and string object for NXdata tag in NeXus reader

### PR DESCRIPTION
**Pull request by ptim0626**
This PR (originally hyperspy/hyperspy#3137) is a small improvement on the NeXus reader, allowing it to recognise NXdata tag as both byte and string object. This increases flexibility and avoid missing genuine NXdata dataset when its tag is saved as a string object.

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/rosettasciio/blob/main/upcoming_changes/README.rst)),
- [x] add test,
- [x] ready for review.

### CODE REVIEW

**Review by ericpre — COMMENTED** [review]
Could you please add a test?

File: rsciio/nexus/_api.py [hunk 1]
```diff
@@ -745,7 +745,8 @@ def find_data_in_tree(group, rootname):
                 target = _getlink(group, rootkey, key)
                 if "NX_class" in value.attrs:
                     if (
-                        value.attrs["NX_class"] == b"NXdata"
+                        (value.attrs["NX_class"] == b"NXdata"
+                        or value.attrs["NX_class"] == "NXdata")
```
[the remark this answers is not in the document]
Comment (ptim0626) [reply rsciio/nexus/_api.py:749]: Yes you are right it is more readable

**Comment by ptim0626** [comment]
> Could you please add a test?
Yes it is done now.

**Comment by ericpre** [comment]
pre-commit.ci autofix

**Pull request merged**